vnext/include/ReactUWP/Views/ViewManagerBase.h

```diff
@@ -8,12 +8,76 @@
 #include <XamlView.h>
 #include <folly/dynamic.h>
 #include <yoga/yoga.h>
+#include "ShadowNodeBase.h"
+#include <unordered_map >
+
+#define VIEWMANAGER_IMPL(VMClassName, XAMLType, vmType) \
+  const char *GetName() const override {                \
+    return VMClassName;                                 \
+  }                                                     \
+  using TargetXamlType = XAMLType;                      \
+  using ViewManagerType = vmType;
 
-namespace facebook {
 namespace react {
-struct ShadowNode;
+namespace uwp {
+struct ShadowNodeBase;
 }
-} // namespace facebook
+} // namespace react
+
+template <typename TargetXamlType, typename VMType>
+struct RCTUpdatePropertyHandler {
+  typedef void (VMType::*Type)(
+      react::uwp::ShadowNodeBase *,
+      winrt::impl::com_ref<TargetXamlType>,
+      const std::string &,
+      const folly::dynamic &);
+};
+
+#define VIEWMANAGER_PROPERTY_MAP_BEGIN \
```

Comment:
>VIEWMANAGER_PROPERTY_MAP_BEGIN [](start = 8, length = 30)

It would be nice if could explore using the custom attributes as we do for the NativeModules 2.0. They closely match the Java custom attributes and Objective-C macros.

We are also looking into implementing the DLL based deployment with ABI-safe API. It means that we should be able to create our View Managers using the same API as are going to offer the community developers.